### ISSUE TITLE
Add TypeSafety to `Entity.toObject()`

### ISSUE THREAD

**Issue by SpencerKaiser** [mikro-orm/mikro-orm#4198]
**Is your feature request related to a problem? Please describe.**
`.toObject` does not have solid TypeSafety for the args or on the return type

**Describe the solution you'd like**
Refactor the method to have a signature similar to the following:
```ts
T.toObject(ignoreFields: [keyof T][]) => Omit<EntityDTO<T>, typeof ignoreFields>
```

I believe it's possible to infer the type of `ignoreFields`, but maybe not... if not, maybe a generic argument?

**Comment by B4nan** [comment]
Looks like this does the job just fine, given we don't support dot paths in there, it can be really simple:

```ts
toObject<K extends EntityKey<T>>(ignoreFields?: K[]): Omit<EntityDTO<T>, K>;
```

The `EntityKey` type removes function properties and symbols from keys.

**Comment by B4nan** [comment]
implemented via 18570d7d7e8064c95d5f81a01b57ceeca1653838
